toolchain/sem_ir/inst_fingerprinter.cpp

```diff
@@ -73,11 +73,14 @@ struct Worklist {
       return;
     }
     const auto& entity_name = sem_ir->entity_names().Get(entity_name_id);
-    if (entity_name.bind_index.has_value()) {
-      Add(entity_name.bind_index);
+    if (entity_name.bind_index().has_value()) {
+      Add(entity_name.bind_index());
       // Don't include the name. While it is part of the canonical identity of a
       // compile-time binding, renaming it (and its uses) is a compatible change
       // that we would like to not affect the fingerprint.
+      //
+      // Also don't include the `is_template` flag. Changing that flag should
+      // also be a compatible change from the perspective of users of a generic.
```

Comment:
FWIW, I might've misunderstood our conversation, I thought you were saying this _should_ be part of the fingerprint.

Reply:
This should be -- and is -- part of the input that we use when forming a canonical `EntityName`, from `EntityNameStore::MakeCanonical`. (Two parameters of different entities shouldn't be canonicalized to the same thing if one is a template parameter and the other is not, because the templateness affects the constant value. But it's OK for two entities in different versions of the same source code to have the same fingerprint if one has a template parameter and the other has a generic parameter, and it'd be nice if that change didn't affect the fingerprint of the entity and therefore the fingerprint of all users of the entity.)